akka-docs/src/main/paradox/typed/testing.md

```diff
@@ -287,3 +179,130 @@ Scala
 
 Java
 :   @@snip [ManualTimerExampleTest.scala](/akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/ManualTimerExampleTest.java) { #manual-scheduling-simple }
+
+
+## Synchronous behavior testing
+
+The `BehaviorTestKit` provides a very nice way of unit testing a `Behavior` in a deterministic way, but it has
+some limitations to be aware of.
+
+Certain @apidoc[Behavior]s will be hard to test synchronously and the `BehaviorTestKit` doesn't support testing of
+all features. In those cases the @ref:[asynchronous ActorTestKit](#asynchronous-testing) is recommended. Example of
+limitations:
+
+* Spawning of @scala[`Future`]@java[`CompletionStage`] or other asynchronous task and you rely on a callback to
+  complete before observing the effect you want to test.
+* Usage of scheduler or timers not supported.
+* `EventSourcedBehavior` can't be tested.
+* Interactions with other actors must be stubbed.
+* Blackbox testing style.
+
+The `BehaviorTestKit` will be improved and some of these problems will be removed but it will always have limitations.
+
```

Comment:
moved from above, but the paragraphs regarding the limitations are new

Reply:
Good to list limitations here :+1: